docs/nsm_api.md

```diff
@@ -160,14 +160,30 @@ message RemoteConnectionRequest {
 }
 
 /* 
- * RemoteMechnismRequest defines a request for a particular remote mechanism
+ * RemoteMechanism defines a request for a particular remote mechanism
  *
- * RemoteMechanismType - type of remote mechanism being requested
- * constraints - constraints on the remote mechanism.
+ * type   - type of remote mechanism being requested
+ *
+ * labels - a set of key value pairs that can be used
+ *                       to describe parameters or constraints on parameters
+ *                       for the RemoteMechanism
+ *
+ * Example:
+ *          VXLAN - would have labels src_ip, src_port, dst_ip, dest_port, vni, vnis
+ *                  when NSM1 sends a RemoteConnectionRequest to NSM2, it would specify
+ *                  src_ip, src_port, and vnis.  vnis is a range constructed list of
+ *                  vnis that could be used by NSM2 with this src_ip and src_port.
+ *                  vnis = 1-100, 200-450
+ *                  when NSM2 sends a RemoteConnectionReply, it would specify
+ *                  use the src_ip, src_port from the RemoteConnectionRequest,
+ *                  specify its own dst_ip, dst_port, and pic from the presented vnis
+ *                  the vni to be used
+ *                  vni=12
+ *                  
  */
 message RemoteMechanismRequest {
     RemoteMechanismType type = 1;
-    repeated RemoteMechanismConstraint constraints = 2
+    map<string,string> labels = 2
```

Comment:
I think we should have a message RemoteMechanism, which consists of RemoteMechanismType and map with constraints specific to this RemoteMechanism. In request we should send a slice of RemoteMechanisms. In this case the logic of parsing it becomes more clear. 
something like this:
```
message RemoteMechanism {
    RemoteMechanismType type = 1;
    map<string,string> constraints = 2;
}
message RemoteMechanismRequest {
    repeated common.RemoteMechanism remote_mechanism = 1;
}
```

Reply:
I forgot to change RemoteMechanismRequest here to RemoteMechanism.  Fixed in latest push.